syntax = "proto3";

package Environment;

service Environment
{
	rpc handshake_and_validate(HandshakeRequest) returns (HandshakeResponse){}
	rpc get_env_data(EnvDataRequest) returns (EnvDataResponse){}
	rpc set_action_data(ActionRequest) returns (ActionResponse){}
}


message HandshakeRequest
{
	int32 n_observations = 1;
	int32 n_actions = 2;
}

message HandshakeResponse
{
	int32 id = 1;
	int32 status = 2;
}


message EnvDataRequest
{
	int32 id = 1;
}

message EnvDataResponse
{
	bytes observation = 1;
	float reward = 2;
	bool terminated = 3;
	bool truncated = 4;
	int32 episode = 5;
}


message ActionRequest
{
	int32 id = 1;
}

message ActionResponse
{
	int32 status = 1;
}


message EnvData {
	bytes observation = 1;
	float reward = 2;
	bool terminated = 3;
	bool truncated = 4;
	int32 episode = 5;
}